xbmc/utils/CPUInfo.cpp

```diff
@@ -419,6 +420,30 @@ CCPUInfo::CCPUInfo(void)
       }
     }
     fclose(fCPUInfo);
+    // new socs use the sysfs soc interface to describe the hardware
+    if (SysfsUtils::Has("/sys/bus/soc/devices/soc0"))
+    {
+      std::string machine, family, soc_id, revision;
+      if (SysfsUtils::Has("/sys/bus/soc/devices/soc0/machine"))
+        SysfsUtils::GetString("/sys/bus/soc/devices/soc0/machine", machine);
+      if (SysfsUtils::Has("/sys/bus/soc/devices/soc0/family"))
+        SysfsUtils::GetString("/sys/bus/soc/devices/soc0/family", family);
+      if (SysfsUtils::Has("/sys/bus/soc/devices/soc0/soc_id"))
+        SysfsUtils::GetString("/sys/bus/soc/devices/soc0/soc_id", soc_id);
+      if (SysfsUtils::Has("/sys/bus/soc/devices/soc0/revision"))
+        SysfsUtils::GetString("/sys/bus/soc/devices/soc0/revision", revision);
+      if (!machine.empty())
+        m_cpuHardware = machine;
+      if (!family.empty() && !soc_id.empty())
+      {
+        if (m_cpuModel.empty())
+          m_cpuModel = family + " " + soc_id;
+        else
+          m_cpuModel += "(" + family + " " + soc_id + ")";
```

Comment:
I tested this earlier and I'd remove the extra brackets here as the /sys/soc_id on Amlogic outputs `GXBB (S905-H)` so the compound GUI entry shows `ARMv8 Processor rev 4 (v8l) (Amlogic Meson GXBB (S905-H))` which is ugly. The (IMHO better) alternative would be to add another line item for `soc: Amlogic Meson GXBB (S905-D)` that's separate from the CPU details.

Reply:
@chewitt I agree for the extra brackets, for the SoC entry it will need more work, and it could even split into "SoC Family" and "SoC Model"

Reply:
I added a new "SoC" line and moved the soc family and id to this identifier, I also dropped the revision which is too detailed and not related to the board revision.